scripts/process/geom_dataframe.R

```diff
@@ -0,0 +1,57 @@
+#' from a sp (will be sf) object, optionally clip and optionally 
+#' mutate data into svg element attributes
+#' 
+#' @details 
+#' This function takes a single required dependency, which is `spatial_data`
+#' Presently, `spatial_data` is Spatial{X}DataFrame, but this will be updated to `sf`
+#' if `eval_args` are present, they are used to convert the `data` slot by evaluating
+#' simple functions w/ variables in `data` (skipped if `eval_args` isn't present)
+#' if `clip_box` is present as a dependency, it is used to clip the geometries in 
+#' `spatial_data`
+#' See `eval_data.frame` function 
+process.geom_dataframe <- function(viz){
+  deps <- readDepends(viz)
+  checkRequired(deps, 'spatial_data') # spatial data is the only required depends
+  
+  geom_dataframe <- deps[['spatial_data']]
+  
+  if (!is.null(deps[['clip_box']])){
+    # remove all polygons that are outside the box
+    # trim all polgons that overlap the box
+    message('process.geom_dataframe clipping is not implemented yet')
```

Comment:
See comments in https://github.com/USGS-VIZLAB/vizstorm/issues/23 for how this would work.